### PR TITLE
Fix autoinstall snippets

### PR DESCRIPTION
**Pull request by heroin-moose**
Highlights:
- Fix autoinstall snippets
- Add missing breeds to autoinstall_start
- Execute nopxe prior to runpost
- Make autoinstall snippets more generic
- Further reduce code duplication

### CODE REVIEW

**Comment by heroin-moose** [comment]
For some reason CentOS8 build failed with the following:

```
 Problem: cannot install the best candidate for the job
  - nothing provides rpm = 4.14.3-14.el8_4 needed by rpm-build-4.14.3-14.el8_4.x86_64
(try to add '--skip-broken' to skip uninstallable packages or '--nobest' to use not only best candidate packages)
```

**Review by SchoolGuy — APPROVED** [review]
LGTM in my eyes. Havn't had the time to test it yet but will before the release.

File: autoinstall_snippets/autoinstall_start [hunk 1]
```diff
@@ -1,31 +1,26 @@
...
-        #end if
-    #end if
-#else if $profile_name != ''
+#if $object_type != ''
     ## RUN PRE TRIGGER
-    #if $run_install_triggers in [ "1", "true", "yes", "y" ]
+    #if $run_install_triggers == "True"
         #if $breed == 'redhat'
-            #set runpre = "\ncurl \"http://%s/cblr/svc/op/trig/mode/pre/%s/%s\" -o /dev/null" % (srv, "profile", profile_name)
-        #else if $breed == 'suse'
```
Comment (vzhestkov) [inline autoinstall_snippets/autoinstall_start:25]: Now it seems that `-O` is the default for `suse`. Just checked with SLE12SP5 and it's not handling `-O`, just `-o`. Could it be the problem?

File: autoinstall_snippets/autoinstall_done [hunk 2]
```diff
@@ -11,80 +21,42 @@
...
+            #set runpost = "\ncurl \"http://%s/cblr/svc/op/trig/mode/post/%s/%s\" -o /dev/null" % (srv, object_type, object_name)
         #else if $breed == 'vmware' and $os_version == 'esx4'
-            #set runpost = "\ncurl \"http://%s/cblr/svc/op/trig/mode/post/%s/%s\" -o /dev/null" % (srv, "profile", profile_name)
-        #else if $breed == 'vmware'
-            #set runpost = "\nwget \"http://%s/cblr/svc/op/trig/mode/post/%s/%s\" -O /dev/null" % (srv, "profile", profile_name)
-        #else if $breed == 'debian' or $breed == 'ubuntu'
-            #set runpost = "\nwget \"http://%s/cblr/svc/op/trig/mode/post/%s/%s\" -O /dev/null" % (srv, "profile", profile_name)
+            #set runpost = "\ncurl \"http://%s/cblr/svc/op/trig/mode/post/%s/%s\" -o /dev/null" % (srv, object_type, object_name)
+        #else
+            #set runpost = "\nwget \"http://%s/cblr/svc/op/trig/mode/post/%s/%s\" -O /dev/null" % (srv, object_type, object_name)
```
Comment (vzhestkov) [inline autoinstall_snippets/autoinstall_done:56]: Now it seems that -O is the default for suse. Just checked with SLE12SP5 and it's not handling -O, just -o. Could it be the problem?

Comment (heroin-moose) [reply autoinstall_snippets/autoinstall_done:56]: It's not only -o, it's also wget instead of curl.

Comment (vzhestkov) [reply autoinstall_snippets/autoinstall_done:56]: Oh, sorry. Missed it.

File: autoinstall_snippets/autoinstall_start (hunk 1 above)
Comment (vzhestkov) [reply autoinstall_snippets/autoinstall_start:25]: Ignore the previous comment, please.

**Review by vzhestkov — APPROVED** [review]

**Pull request merged**